lib/ueberauth/strategy/github.ex

```diff
@@ -192,11 +192,18 @@ defmodule Ueberauth.Strategy.Github do
 
   defp fetch_user(conn, token) do
     conn = put_private(conn, :github_token, token)
+    # Will be better with Elixir 1.3 with/else
     case OAuth2.AccessToken.get(token, "/user") do
       { :ok, %OAuth2.Response{status_code: 401, body: _body}} ->
         set_errors!(conn, [error("token", "unauthorized")])
       { :ok, %OAuth2.Response{status_code: status_code, body: user} } when status_code in 200..399 ->
-        put_private(conn, :github_user, user)
+        case OAuth2.AccessToken.get(token, "/user/emails") do
+          { :ok, %OAuth2.Response{status_code: status_code, body: emails} } when status_code in 200..399 ->
+            user = Map.put user, "emails", emails
+            put_private(conn, :github_user, user)
+          { :error, _ } -> # Continue on as before
+            put_private(conn, :github_user, user)
+        end
```

Comment:
Would rather to refactor this to use `with` and bump the mix file up to 1.2, but I'll leave that decision up to you. This will be much cleaner with 1.3 so it may not be worth it until then.